### DISCUSSION TITLE
Vertically Integrated Water Vapor

### DISCUSSION THREAD

**Discussion by kbrews** [ufs-community/ufs-srweather-app#980] [Ideas]
Vapor Flux, title should say.  We are interested in using Vertically Integrated Water Vapor Flux as an output of UPP in SRWapp for our work with NOAA HMT.  NCEP GRIB2 doc (Table 4.2.1) calls this variable CWVF, column integrated water vapor flux.     This has been identified as a critical feature for intense rainfall events -- think LLJ and Atmospheric Rivers -- but the variable does not seem to be in the UPP from documentation I can find.  We can get an estimate from pressure-level output, but like PWAT, it'd be better to integrate in the model coordinates.  Before we start modifying code to add this, has anyone already gone down this road?

**Comment by gspetro-NOAA** [comment]
Hello! 

I checked in with one of our UPP code managers, and he is not aware of any updates in the works to incorporate CWVF. However, we're going to check in with the UPP team to see if anyone else has heard of work underway. In the future, it may be best to post UPP-specific questions in the [UPP](https://github.com/NOAA-EMC/UPP) repository, since you'll get more eyes on them from UPP developers. I am going to copy/repost your question there. 

If you do need to add a variable, there is documentation in the [UPP User's Guide](https://upp.readthedocs.io/en/latest/AddNewVariable.html), although it sounds like you may already be familiar with that. 

Typically, UPP changes are incorporated into the SRW App via a hash update, although @MichaelLueken may be able to comment on whether additional updates would be needed. 

Best,
Gillian Petro

**Comment by gspetro-NOAA** [comment]
This question has been reposted in the [UPP Discussions](https://github.com/NOAA-EMC/UPP/discussions/833) as Discussion 833. Additional information is available there from the UPP team.

**Comment by gspetro-NOAA** [comment]
@kbrews CWVF was added to the UPP last week in [PR #880](https://github.com/NOAA-EMC/UPP/pull/880) .
